lib/internal/http2/core.js

```diff
@@ -1086,6 +1086,18 @@ class ClientHttp2Session extends Http2Session {
     }
     return stream;
   }
+
+  ref() {
+    if (this[kSocket]) {
```

Comment:
Can you also check whether `typeof this[kSocket].ref === 'function'`? HTTP/2 supports any stream as the underlying socket, so `.ref` might not be present at that point.

The test file that checks that is `test/parallel/test-http2-generic-streams.js`, if you want to extend that – I think calling `.ref()` there would crash right now?